### DISCUSSION TITLE
Sphinx ignores docstrings in `@declared_attr` methods

### DISCUSSION THREAD

**Discussion by cjw296** [sqlalchemy/sqlalchemy#8682] [Usage Questions] [answered]
Not sure if this is the responsibility of Sphinx or SQLAlchemy but figure since it's a feature of SQLAlchemy I'd ask here first. 
Take this declarative mixin as an example:

```python
class Common:
    
    @declared_attr
    def __tablename__(cls: Type) -> str:
        """
        The table name is, by default, the snake-case version of the class name.
        """
        return de_hump(cls)
```

I included it in my api.rst as follows:

```rst
Mixins
------

.. autoclass:: Common
   :members:
   :special-members:
   :exclude-members: __ne__,__weakref__
```

Other documented methods are shown, but not `__tablename__`. How do I get that `@declared_attr` method to be included?

**Comment by zzzeek** [comment]
IIUC we populate ``__doc__`` on the member but Sphinx keeps changing how it does things, so we can do whatever is needed here. it's just not clear.  i was thinking of #5999 but that has to do with annotations.

**Reply by cjw296** [reply, marked as answer]
Yeah, pragmatically I'm just going to document this in the .rst file:

```rst
.. autoclass:: Common
    :members:
    :special-members:
    :exclude-members: __ne__,__weakref__

    .. https://github.com/sqlalchemy/sqlalchemy/discussions/8682

    .. attribute:: __tablename__

       The table name is, by default, the snake-case version of the class name.

.. autoclass:: Temporal
    :members:

    .. method:: __init__(period: DateTimeRange = None, value_from: datetime = None, value_to: datetime = None, ...)

        This behaves the same as the normal constructor for a mapped object, but the
        :attr:`period` may alternatively be set by passing either or both of
        :attr:`value_from` or :attr:`value_to`.

```

I have to for `__init__` in the other mixin anyway as that docstring isn't picked up either, but I guess that's expected?

**Reply by zzzeek** [reply]
if you have an explicit `__init__` method somewhere that doesn't have a decorator of some kind on it, I dont see why sphinx wouldn't pick that up

**Reply by cjw296** [reply]
Me neither, but 🤷
